x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/DataNodeRequest.java

```diff
@@ -115,6 +122,10 @@ public String[] indices() {
     @Override
     public IndicesRequest indices(String... indices) {
         this.indices = indices;
+        if (Arrays.asList(indices).contains(NO_INDEX_PLACEHOLDER)) {
```

Comment:
@nik9000 just a heads up: discussed with Jake and its cleaner to move this here since that's what gets called in index resolution with `NO_INDEX_PLACEHOLDER` (as opposed to keep in the get). LMK if you have objections!

Reply:
nit: should we check the length of the String[] for  == 1 && contains no index place holder as an optimization ? It could short circuit (or remove) the need to convert to a List as well as a (presumable) O(N) check where N is all the allowed indices per pattern which could be quite a large list for large clusters with many indices.

Reply:
good call -- it's actually `NO_INDICES_OR_ALIASES_ARRAY` that gets passed in which is two elements. So I'm going with `Arrays.equals(indices, NO_INDICES_OR_ALIASES_ARRAY) || Arrays.asList(indices).contains(NO_INDEX_PLACEHOLDER)` -- `Arrays.equals` does the optimizations under the hood and then we fall back to the expensive check just to be more future-proof.